include/flatbuffers/base.h

```diff
@@ -218,11 +211,17 @@
   #define __supress_ubsan__(type)
 #endif
 
-// This is constexpr function used for checking compile-time constants.
-// Avoid `#pragma warning(disable: 4127) // C4127: expression is constant`.
-template<typename T> FLATBUFFERS_CONSTEXPR inline bool IsConstTrue(T t) {
-  return !!t;
-}
+
+// Supress false positive warning C4127: expression is constant`.
+// This false warning generated under MSVC201/MSVC2013 only.
+// The suppress	pushes the current state of the pragma on the stack,
+// disables the specified warning for the next line,
+// and then pops the warning stack so that the pragma state is reset.
+#if defined _MSC_VER && _MSC_VER < 1900
```

Comment:
In general, this PR seems to be fixing a LOT of unrelated things, certainly unrelated to the title of the PR. While I appreciate you're cleaning up code as you find them, this all would be a lot easier to review if util.h changes were split from other #include optimisations, const related changes, etc.

Reply:
I agree